react/features/remote-video-menu/components/web/MuteEveryoneDialog.js

```diff
@@ -0,0 +1,137 @@
+/* @flow */
+
+import React from 'react';
+
+import {
+    AUDIO_MUTE,
+    createToolbarEvent,
+    sendAnalytics
+} from '../../../analytics';
+import { Dialog } from '../../../base/dialog';
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import { setAudioMuted } from '../../../base/media';
+import UIEvents from '../../../../../service/UI/UIEvents';
+
+import AbstractMuteRemoteParticipantDialog, {
+    type Props as AbstractProps
+} from '../AbstractMuteRemoteParticipantDialog';
+
+declare var APP: Object;
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link MuteEveryoneDialog}.
+ */
+type Props = AbstractProps & {
+
+    /**
+     * The ID of the local participant.
+     */
+    localParticipantId: string,
+
+    /**
+     * The IDs of the remote participants to be muted.
+     */
+    remoteParticipantIds: Array<string>,
+
+     /**
+     * The IDs of the remote participants to exclude from being muted.
+     */
+    exclude: Array<string>,
+};
+
+/**
+ * A React Component with the contents for a dialog that asks for confirmation
+ * from the user before muting a remote participant.
+ *
+ * @extends Component
+ */
+class MuteEveryoneDialog extends AbstractMuteRemoteParticipantDialog<Props> {
+    static defaultProps = {
+        exclude: [],
+        muteLocal: false
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t } = this.props;
+
+        return (
+            <Dialog
+                okKey = 'dialog.muteMultipleParticipantButton'
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.muteMultipleParticipantTitle'
+                width = 'small'>
+                <div>
+                    { t('dialog.muteMultipleParticipantBody') }
+                </div>
+            </Dialog>
+        );
+    }
+
+    _onSubmit: () => boolean;
+
+    /**
+     * Callback to be invoked when the value of this dialog is submitted.
+     *
+     * @returns {boolean}
+     */
+    _onSubmit() {
+        const {
+            dispatch,
+            exclude,
+            localParticipantId,
+            remoteParticipantIds
+        } = this.props;
+
+        remoteParticipantIds
+            .filter(id => !exclude.includes(id))
+            .forEach(id => {
+                if (id === localParticipantId) {
+                    const audioMuted = true;
+
+                    sendAnalytics(createToolbarEvent(AUDIO_MUTE, { enable: audioMuted }));
```

Comment:
This whole thing is totally copy pasted from the audio mute button in the toolbar....

I guess you can find better solution.

Reply:
I am doing invidual mutes in a loop - what is the problem with sending analytics individually?

Reply:
There isn't a problem with sending individual analytics. The problem is that  you've copy pasted relatively big chunk of code from other parts of the project. The least you can do is to create a function which will be used in both places. WDYT?

Reply:
i guess since this is now consolidated in an action, this is also resolved